src/components/pages/Onboarding/Onboarding.tsx

```diff
@@ -68,35 +75,35 @@ type OnboardingState = {
 class OnboardingView extends React.Component<OnboardingProps, OnboardingState> {
   dropdown: any;
   steps: OnboardingStep[];
-  deeplinkInitialized: boolean;
   videoToken: string;
+  lastInviteVideoToken?: string;
 
   constructor(props: OnboardingProps) {
     super(props);
     this.steps = [];
-    this.deeplinkInitialized = false;
     this.videoToken = generateToken();
     this.state = {
       step: OnboardingStep.SPLASH
     };
-
-    if (this.props.hasValidInviteToken) {
-      this.initializeDeeplinkingState();
-    }
+    this.initializeDeeplinkingState();
   }
 
   /**
-   * If deeplinking params are present, makes sure they are valid, and fills in
-   * the associated video download url into state.
+   * Idempotently initializes deeplinking state. If deeplinking params are present,
```

Comment:
Sounds like it's not really initializing it, but something else (since initialization sounds like the kind of thing you do once). Might be a little confusing

Reply:
It initializes once per token change though. Any suggestions on what to change it to?